Add Youtube repositories

sheetal-g9 opened the pull request:
First steps for issues #119 
Changes:Add playlist and video repositories ,fetching data with api and storing them in playlist and video models.

Commit:
add playlist and video repositories

Commit:
add playlist model class YoutubePlaylist.java

Commit:
add video model class YoutubeVideo.java

Commit:
minor changes

dr0pdb reviewed (changes requested):
Go through the codacy issues and fix them. Also look at why the build is failing and fix it.

Commit:
add youtube to build.gradle and add api.jar

Commit:
minor changes: add access specifier and remove explicit package name

BVShe001 commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/VideoRepository.java:
```diff
@@ -0,0 +1,110 @@
+                }
+
+                @Override
+                public void onComplete() {
```
Observer requires this method to be implemented.

BVShe001 commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/PlaylistRepository.java:
```diff
@@ -0,0 +1,125 @@
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(new Observer<List<YoutubePlaylist>>() {
+                @Override
+                public void onSubscribe(Disposable d) {
```
Observer requires this method to be implemented.

BVShe001 commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/PlaylistRepository.java:
```diff
@@ -0,0 +1,125 @@
+                }
+
+                @Override
+                public void onComplete() {
```
Observer requires this method to be implemented.

srv-twry replied:
For empty body add a comment such as `// nothing here`.

BVShe001 commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/VideoRepository.java:
```diff
@@ -0,0 +1,110 @@
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(new Observer<List<YoutubeVideo>>() {
+                @Override
+                public void onSubscribe(Disposable d) {
```
Observer requires this method to be implemented.

dr0pdb commented:
@BVShe001 I know that those empty methods are required but Codacy is suggesting you to **document** it not remove it. Add a comment like `nothing here` inside the method body.

Commit:
minor changes in name

Commit:
add comment in empty methods

sheetal-g9 commented:
Build is failing because it is not getting api key which I have abstracted as in :
[ how to hide](https://medium.com/code-better/hiding-api-keys-from-your-android-repository-b23f5598b906)
It is building perfectly in my Android Studio.

dr0pdb commented:
@BVShe001 
1. If we're going to let the build fail now and merge this one then all the future builds will fail. Then what's the point of these CI's?
2. Search for how to setup demo api-key on Travis so that the build doesn't fail.

dr0pdb commented on .travis.yml:
```diff
@@ -1,27 +1,30 @@
+  - tools
+  - android-27
+  - build-tools-27.0.3
+  - platform-tools
```
What have you done here? Don't make unnecessary changes.

dependabot[bot] committed:
build(deps): bump facebook-android-sdk from 4.38.1 to 4.39.0 (#217)

Bumps [facebook-android-sdk](https://github.com/facebook/facebook-android-sdk) from 4.38.1 to 4.39.0.
- [Release notes](https://github.com/facebook/facebook-android-sdk/releases)
- [Commits](https://github.com/facebook/facebook-android-sdk/compare/sdk-version-4.38.1...sdk-version-4.39.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
remove build config field for youtube_api_key

ravi-kishan committed:
feat: Add facebook post Model (#219)

ravi-kishan committed:
feat: Implement fb post DAO and deserialization (#220)

dr0pdb commented on app/build.gradle:
```diff
@@ -110,7 +110,7 @@ dependencies {
     implementation "com.squareup.retrofit2:converter-jackson:$rootProject.retrofitVersion"
     implementation "com.squareup.retrofit2:adapter-rxjava2:$rootProject.retrofitVersion"
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.12.0'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.11.0'
```
Why decrease the version?

dr0pdb commented on app/build.gradle:
```diff
@@ -171,9 +171,27 @@ dependencies {
     })
     testImplementation 'junit:junit:4.12'
-    testImplementation 'org.mockito:mockito-inline:2.23.4'
+    testImplementation 'org.mockito:mockito-inline:2.23.0'
```
Why decrease it?

dr0pdb commented on app/build.gradle:
```diff
@@ -77,9 +80,6 @@ check {
 
-    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
```
Why remove it?

sheetal-g9 replied:
It is redundant.Same dependency is used at line :170 with a newer version 3.0.1

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/models/YoutubePlaylist.java:
```diff
@@ -0,0 +1,43 @@
+
+    public long getItemCount() {
+        return itemCount;
+    }
```
Please take a look at other model classes. These getters, setters, constructors can easily be done using annotations such as `@data`, `@allargsconstructor` etc. Use them.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/models/YoutubeVideo.java:
```diff
@@ -0,0 +1,34 @@
+
+    public String getVideoId() {
+        return video_id;
+    }
```
Check other models.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/PlaylistRepository.java:
```diff
@@ -0,0 +1,126 @@
+
+public class PlaylistRepository {
+    public GoogleAccountCredential googleAccountCredential;
+    private String channelName = "GoogleDevelopers";
```
What is this?

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/PlaylistRepository.java:
```diff
@@ -0,0 +1,126 @@
+    public void getDataFromAPI() {
+        HttpTransport transport = AndroidHttp.newCompatibleTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+        service = new YouTube.Builder(
```
Use Dagger for it. Create an instance of YouTube builder and inject it in the Repository as a dependency just like `GoogleAccountCredential`.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/repositories/VideoRepository.java:
```diff
@@ -0,0 +1,111 @@
+    public void getDataFromAPI() {
+        HttpTransport transport = AndroidHttp.newCompatibleTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+        service = new YouTube.Builder(
```
Use Dagger.

Commit:
add lombok for for constructor getter and setter

Commit:
add Youtube Module for Dependency Injection

Commit:
Add dependency injection for youtube service dependency

Add dependency injection for youtube service dependency

build(deps): bump facebook-android-sdk from 4.38.1 to 4.39.0 (#217)

Bumps [facebook-android-sdk](https://github.com/facebook/facebook-android-sdk) from 4.38.1 to 4.39.0.
- [Release notes](https://github.com/facebook/facebook-android-sdk/releases)
- [Commits](https://github.com/facebook/facebook-android-sdk/compare/sdk-version-4.38.1...sdk-version-4.39.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

feat: Add facebook post Model (#219)

feat: Implement fb post DAO and deserialization (#220)

build(deps): bump robolectric from 4.0.2 to 4.1 (#218)

Bumps [robolectric](https://github.com/robolectric/robolectric) from 4.0.2 to 4.1.
- [Release notes](https://github.com/robolectric/robolectric/releases)
- [Commits](https://github.com/robolectric/robolectric/compare/robolectric-4.0.2...robolectric-4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

minor changes :change directory

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/YoutubeModule.java:
```diff
@@ -0,0 +1,66 @@
+    @Provides
+    @Singleton
+    PlaylistRepository providesPlaylistRepository() {
+        return new PlaylistRepository();
```
Don't use DI for providing repository. Revert this back to what it was.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/YoutubeModule.java:
```diff
@@ -0,0 +1,66 @@
+    @Provides
+    @Singleton
+    VideoRepository providesVideoRepository() {
+        return new VideoRepository();
```
Same here.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/YoutubeModule.java:
```diff
@@ -0,0 +1,66 @@
+    @Singleton
+    HttpTransport providesHttpTransport(){
+        return AndroidHttp.newCompatibleTransport();
+    }
```
Add an extra line!

dependabot[bot] committed:
build(deps): bump robolectric from 4.0.2 to 4.1 (#218)

Bumps [robolectric](https://github.com/robolectric/robolectric) from 4.0.2 to 4.1.
- [Release notes](https://github.com/robolectric/robolectric/releases)
- [Commits](https://github.com/robolectric/robolectric/compare/robolectric-4.0.2...robolectric-4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>
